File: Problems & Solutions/B/Balanced string/15 December 2022/C++/Balanced string.cpp
//{ Driver Code Starts
// Initial template for c++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution {
  public:
    string BalancedString(int N) {
        string ans = "", full = "";
        for (char ch = 'a'; ch <= 'z'; ch++) full += ch;
        int a = N / 26, b = N % 26, i;
        while (a--)
            ans += full;
        if (N % 2 == 0) {
            for (i = 0; i < b / 2; i++) ans += (char)('a' + i);
            for (i = 26 - b / 2; i < 26; i++) ans += (char)('a' + i);
        }
        else {
            int SumOfdigits = 0, temp = N;
            while (temp) {
                SumOfdigits += temp % 10;
                temp /= 10;
            }
            if (SumOfdigits % 2 == 0) {
                for (i = 0; i < (b + 1) / 2; i++) ans += (char)('a' + i);
                for (i = 26 - b / 2; i < 26; i++) ans += (char)('a' + i);
            }
            else {
                for (i = 0; i < b / 2; i++) ans += (char)('a' + i);
                for (i = 26 - (b + 1) / 2; i < 26; i++) ans += (char)('a' + i);
            }
        }
        return ans;
    }
};


//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        cin >> N;
        Solution ob;
        cout << ob.BalancedString(N) << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Problems & Solutions/B/Balanced string/15 December 2022/Python/Balanced string.py
class Solution:
    def BalancedString(self,N):
        ans=""
        full="abcdefghijklmnopqrstuvwxyz"
        a=N//26
        b=N%26
        for _ in range(a):
            ans=ans+full
        if(N%2==0):
            for i in range(b//2):
                ans=ans+full[i]
            for i in range(26-b//2,26):
                ans=ans+full[i]
        else:
            SumOfDigits=0
            temp=N
            while(temp!=0):
                SumOfDigits=SumOfDigits+(temp)%10
                temp=temp//10  
            if(SumOfDigits%2):
                for i in range((b-1)//2):
                    ans=ans+full[i]
                for i in range(26-(b+1)//2,26):
                    ans=ans+full[i]    
            else:   
                for i in range((b+1)//2):
                    ans=ans+full[i]
                for i in range(26-(b-1)//2,26):
                    ans=ans+full[i]
        return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__=='__main__':
    t=int(input())
    for _ in range(t):
        N=int(input())
        
        ob=Solution()
        print(ob.BalancedString(N))
# } Driver Code Ends